src/main/java/ch/njol/skript/expressions/ExprWeather.java

```diff
@@ -12,76 +12,92 @@
 import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
 import org.bukkit.World;
+import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
 import org.bukkit.event.weather.ThunderChangeEvent;
 import org.bukkit.event.weather.WeatherChangeEvent;
 import org.bukkit.event.weather.WeatherEvent;
 import org.jetbrains.annotations.Nullable;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Weather")
-@Description("The weather in the given or the current world.")
-@Examples({"set weather to clear",
-		"weather in \"world\" is rainy"})
+@Description({
+	"The weather of a world or player.",
+	"Clearing or resetting the weather of a player will make the player's weather match the weather of the world.",
+	"Clearing or resetting the weather of a world will make the weather clear."
+})
+@Examples({
+	"set weather to clear",
+	"weather in \"world\" is rainy",
+	"reset custom weather of player",
+	"set weather of player to clear"
+})
 @Since("1.0")
 @Events("weather change")
-public class ExprWeather extends PropertyExpression<World, WeatherType> {
+public class ExprWeather extends PropertyExpression<Object, WeatherType> {
+
 	static {
-		Skript.registerExpression(ExprWeather.class, WeatherType.class, ExpressionType.PROPERTY, "[the] weather [(in|of) %worlds%]", "%worlds%'[s] weather");
+		Skript.registerExpression(ExprWeather.class, WeatherType.class, ExpressionType.PROPERTY,
+			"[the] [custom|client] weather [(in|of) %worlds/players%]",
+			"%worlds/players%'[s] [custom|client] weather");
 	}
 
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
-		setExpr((Expression<World>) exprs[0]);
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
+		setExpr(exprs[0]);
 		return true;
 	}
 
 	@Override
-	protected WeatherType[] get(Event event, World[] source) {
-		return get(source, world -> {
-			if (getTime() >= 0 && event instanceof WeatherEvent weatherEvent
-				&& world.equals(weatherEvent.getWorld()) && !Delay.isDelayed(event))
-				return WeatherType.fromEvent(weatherEvent);
-			else
+	protected WeatherType @Nullable [] get(Event event, Object[] source) {
+		World eventWorld = event instanceof WeatherEvent weatherEvent ? weatherEvent.getWorld() : null;
+        return get(source, object -> {
+			if (object instanceof Player player) {
+				return WeatherType.fromPlayer(player);
+			} else if (object instanceof World world) {
+				if (eventWorld != null && world.equals(eventWorld) && getTime() >= 0 && !Delay.isDelayed(event))
```

Comment:
Delay check not needed when the time states have a set event related.
```suggestion
				if (eventWorld != null && world.equals(eventWorld) && getTime() >= 0)
```